tests/pr_test.go

```diff
@@ -178,3 +178,87 @@ func TestWithExistingKP(t *testing.T) {
 	}
 
 }
+
+func TestRunUpgradeExistingKP(t *testing.T) {
+	t.Parallel()
+
+	// ------------------------------------------------------------------------------------
+	// Provision KP first
+	// ------------------------------------------------------------------------------------
+
+	prefix := fmt.Sprintf("kp-ut-%s", strings.ToLower(random.UniqueId()))
+	realTerraformDir := "./resources/kp-instance"
+	tempTerraformDir, _ := files.CopyTerraformFolderToTemp(realTerraformDir, fmt.Sprintf(prefix+"-%s", strings.ToLower(random.UniqueId())))
+	region := "us-south"
+
+	// Verify ibmcloud_api_key variable is set
+	checkVariable := "TF_VAR_ibmcloud_api_key"
+	val, present := os.LookupEnv(checkVariable)
+	require.True(t, present, checkVariable+" environment variable not set")
+	require.NotEqual(t, "", val, checkVariable+" environment variable is empty")
+
+	logger.Log(t, "Tempdir: ", tempTerraformDir)
+	existingTerraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: tempTerraformDir,
+		Vars: map[string]interface{}{
+			"prefix": prefix,
+			"region": region,
+		},
+		// Set Upgrade to true to ensure latest version of providers and modules are used by terratest.
+		// This is the same as setting the -upgrade=true flag with terraform.
+		Upgrade: true,
+	})
+
+	terraform.WorkspaceSelectOrNew(t, existingTerraformOptions, prefix)
+	_, existErr := terraform.InitAndApplyE(t, existingTerraformOptions)
+	if existErr != nil {
+		assert.True(t, existErr == nil, "Init and Apply of temp existing resource failed")
+	} else {
+
+		// ------------------------------------------------------------------------------------
+		// Upgrade test for watsonx DA passing in existing KP details
+		// ------------------------------------------------------------------------------------
+
+		options := testhelper.TestOptionsDefault(&testhelper.TestOptions{
+			Testing:      t,
+			TerraformDir: rootDaDir,
+			Prefix:       "existing-kp-upg",
+			IgnoreDestroys: testhelper.Exemptions{ // Ignore for consistency check
+				List: []string{
+					"module.configure_user.null_resource.configure_user",
+					"module.configure_user.null_resource.restrict_access",
+				},
+			},
+			IgnoreUpdates: testhelper.Exemptions{ // Ignore for consistency check
+				List: []string{
+					"module.configure_user.null_resource.configure_user",
+					"module.configure_user.null_resource.restrict_access",
+				},
+			},
+			TerraformVars: map[string]interface{}{
+				"location":            validRegions[rand.Intn(len(validRegions))],
+				"resource_group_name": prefix,
+				"provider_visibility": "public",
+				"cos_kms_crn":         terraform.Output(t, existingTerraformOptions, "key_protect_crn"),
+				"cos_kms_key_crn":     terraform.Output(t, existingTerraformOptions, "kms_key_crn"),
```

Comment:
To enable encryption, you need to set `enable_cos_kms_encryption = true`. The fact that the test passed even when passing values for `cos_kms_crn` and `cos_kms_key_crn` makes me think we should add variable validation to the DA code to ensure that if user does pass values for these, that `enable_cos_kms_encryption` has to be set to true

Reply:
It also looks like `TestWithExistingKP` also needs this update

Reply:
@ocofaigh Added `enable_cos_kms_encryption = true` , still the test passed.